lib/puppet/provider/rabbitmq_cli.rb

```diff
@@ -1,19 +1,36 @@
 class Puppet::Provider::RabbitmqCli < Puppet::Provider
   initvars
 
-  def self.rabbitmq_command(name, binary)
-    path = Puppet::Util.which(binary) || "/usr/lib/rabbitmq/bin/#{binary}"
-    home_tmp_command name, path
+  def self.append_to_path(dir)
+    path = get_env 'PATH'
+    # Don't append to the PATH if the directory is already in it. Otherwise, if
+    # multiple providers run in the same process it may result in the
+    # environment being modified multiple times.
+    return if path.split(File::PATH_SEPARATOR).include? dir
+
+    set_env 'PATH', [path, dir].join(File::PATH_SEPARATOR)
   end
+  private_class_method :append_to_path
+
+  # On most platforms, the RabbitMQ CLI programs are available in the PATH under
+  # /usr/sbin. On some older platforms (CentOS 6), they are only available at
+  # /usr/lib/rabbitmq/bin. We can't detect which because at the time this file
+  # is evaluated, RabbitMQ might not yet be installed. However, if a command is
+  # specified by name (instead of absolute path), Puppet searches the PATH
+  # before each execution of the command. Append /usr/lib/rabbitmq/bin to the
+  # end of the PATH so that Puppet will look there last at the time a command is
+  # executed. This is the best I can come up with short of fragile meta-
+  # programming with Puppet internals.
+  append_to_path '/usr/lib/rabbitmq/bin'
 
   def self.home_tmp_command(name, path)
     has_command name, path do
       environment HOME: '/tmp'
```

Comment:
Can't you just set PATH in `environment`, or does Puppet override it?

Reply:
Would be nice, but the `which` call is separate 😞 

https://github.com/puppetlabs/puppet/blob/111441de51724ba054cb122aece8324d173c3a8e/lib/puppet/provider/command.rb#L19-L24

The `environment` gets passed through `@options`.

(Don't know why the GitHub code embedding in comments is not working today)